pkg/segment/reader/record/recordreader.go

```diff
@@ -57,6 +57,14 @@ func GetRecordsFromSegment(segKey string, vTable string, blkRecIndexes map[uint1
 			return nil, allCols, errors.New("failed to get column names for segkey in rotated and unrotated files")
 		}
 	}
+
+	// if len(allColsInAggs) > 0, then we need to intersect the allCols with allColsInAggs
+	// this is because we only need to read the columns that are present in the aggregators.
+	// if len(allColsInAggs) == 0, then we ignore this step, as this would mean that the
+	// query does not have a stats block, and we need to read all columns.
+	if len(allColsInAggs) > 0 {
```

Comment:
nit: this could be a good spot to use `utils.Option[T]` instead. Then instead of `if len(allColsInAggs) > 0 {` you'd do:
```
if allCols, ok := allColsInAggs.Get(); ok {
```
(where `allColsInAggs` is now a `Option[map[string]struct{}]` instead of a `map[string]struct{}`). And in this case, you'd still do the intersection if it was a non-nil map of zero elements.

However, you can ignore this and keep what you have if you want

Reply:
Got it. But that would mean then I need to change the type of `allColsInAggs` from where I have first initialized this variable right? 

For now I would like to keep this as it is. But for future PRs I will use `utils.Option[T]`.